lib/client.js

```diff
@@ -5,6 +5,9 @@ const NodeCache = require('node-cache')
 const log = require('loglevel').getLogger('nypl-data-api-client')
 const OAuth = require('oauth')
 
+const TokenExpirationError = require('./errors').TokenExpirationError
+const TokenRefreshError = require('./errors').TokenRefreshError
+
 class Client {
 
   constructor (opts) {
```

Comment:
This has sent me down a delightful ES6 default parameter / destructuring assignment rabbit hole. Thank you.

A common pattern I use a lot is to pass an `options` hash as final param to a function. The default parameter facility (`function foo (param1 = 'default param value', options = {}) {}`) doesn't allow me to set defaults for members of the `options` hash, but the use of destructuring assignment above makes something like this possible:

```
function foo (param1, { opt1 = 'opt1 default', opt2 = 'opt2 default' } = {}) {
  this.param1 = param1
  this.options = { opt1, opt2 }
}
```
That should do what I want. But because it's a destructured assignment, I don't get a reference to the `options` param, so if I want a `this.options` instance var, I have to build `this.options` in the constructor using all of those option keys. In effect I have to maintain the options hash in two different places: once in the function definition to establish defaults and again when I actually assign `this.options`. Also confusing is the right hand of that destructuring assignment, which is one additional place to establish defaults *but only if the parameter is entirely omitted*. Which makes me think maybe this is still clearest - particularly for large `options` params:

```
function foo (param1, options = {}) {
  this.param1 = param1
  this.options = Object.assign({
    opt1: 'opt1 default', 
    opt2: 'opt2 default'
  }, options)
}
```